src/apps/allbridge/positions.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  PositionsHook,
+  AppTokenPositionDefinition,
+  UnknownAppTokenError,
+  TokenDefinition,
+  ContractPositionDefinition,
+} from '../../types/positions'
+import { Address } from 'viem'
+import { DecimalNumber, toDecimalNumber } from '../../types/numbers'
+import BigNumber from 'bignumber.js'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { getAllbridgeTokenInfo } from './api'
+import { poolAbi } from './abis/pool'
+
+const ALLBRIDGE_LOGO =
+  'https://raw.githubusercontent.com/valora-inc/dapp-list/main/assets/allbridgecore.png'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'allbridge',
+      name: 'Allbridge',
+      description: '',
+    }
+  },
+  async getPositionDefinitions(networkId, address) {
+    const allbridgeTokenInfo = (await getAllbridgeTokenInfo({ networkId }))
+      .tokens
+
+    const client = getClient(networkId)
+
+    const [balances, rewards] = await Promise.all([
+      await Promise.all(
+        allbridgeTokenInfo.map(async (tokenInfo) => {
+          return address
+            ? client.readContract({
+                address: tokenInfo.poolAddress,
+                abi: poolAbi,
+                functionName: 'balanceOf',
+                args: [address as Address],
+              })
+            : undefined
+        }),
+      ),
+      await Promise.all(
+        allbridgeTokenInfo.map(async (tokenInfo) => {
+          return address
+            ? client.readContract({
+                address: tokenInfo.poolAddress,
+                abi: poolAbi,
+                functionName: 'pendingReward',
+                args: [address as Address],
+              })
+            : undefined
+        }),
+      ),
+    ])
+
+    return allbridgeTokenInfo.flatMap((tokenInfo, i) => {
```

Comment:
why flatMap? does allbridge return array of arrays?

Reply:
inside the flatMap call, I am returning a list like `[app-token-definition, contract-position-definition]` (the pool and the reward), and so I believe the flatMap makes it so the entire return value is just a list of positions, rather than a list of lists of pairs of positions.